protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -662,6 +638,87 @@ func (rpccs *RPCConsumerServer) sendRelayToProvider(
 	return response.relayResult, response.err
 }
 
+func (rpccs *RPCConsumerServer) getBestResult(finalResult chan *relayResponse, timeout time.Duration, responses chan *relayResponse, numberOfSessions int, chainMessage chainlib.ChainMessage) {
```

Comment:
instead of getting the channel as input just return the value you write to it as a returned value